Change value of css property -webkit-line-box-contain on text track container div

danae404 opened the pull request:
#### 8917b6acfc0d79357c993094fd7b9d8e271ee75e
<pre>
Change value of css property -webkit-line-box-contain on text track container div
<a href="https://bugs.webkit.org/show_bug.cgi?id=298623">https://bugs.webkit.org/show_bug.cgi?id=298623</a>
<a href="https://rdar.apple.com/156714727">rdar://156714727</a>

Reviewed by Tim Nguyen.

On the &lt;div useragentpart=-webkit-text-track-container&gt;, the value of css property
-webkit-line-box-contain should be changed from &quot;block inline-box replaced&quot; to
the default, &quot;block inline replaced&quot;. This will allow us to pass webvtt web platform
test class_font_shorthand.html and probably others as well.

class_font_shorthand.html tests a cue with a nested &lt;c&gt; element, where &lt;c&gt; is given a line height of 18px.
Because of  -webkit-line-box-contain:&quot;block inline-box replaced&quot;, the line height of the cue is not
impacted by the line height of &lt;c&gt;, causing it to differ from the reference file.

* Source/WebCore/Modules/modern-media-controls/controls/text-tracks.css:
(video::-webkit-media-text-track-container):

Canonical link: <a href="https://commits.webkit.org/301157@main">https://commits.webkit.org/301157@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5350906cb191f0d9d7bfc4ec8de144b545e0c0d4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125122 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44789 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35528 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131970 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/074d822f-d995-47f2-9f57-c3cf0244556c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126999 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45480 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53352 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/95245 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/70db6235-7124-4f81-b1b9-09b37db267f6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128076 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36300 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/111884 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75787 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9d41903e-922a-4557-b7ba-af079b5d9480) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/35205 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/30047 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75451 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106065 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30276 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134650 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51932 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/39723 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103715 "Failed to checkout and rebase branch from PR 50515") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108104 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103486 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48830 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27118 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48985 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19596 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51824 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/57608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51191 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54547 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52881 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c55579f827abe9e241bdfc36d2367942a8c7d396. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50515

nt1m reviewed (approved):
Having an isolated test for this would be good

danae404 committed:
Change value of css property -webkit-line-box-contain on text track container div
https://bugs.webkit.org/show_bug.cgi?id=298623
rdar://156714727

Reviewed by Tim Nguyen.

On the <div useragentpart=-webkit-text-track-container>, the value of css property
-webkit-line-box-contain should be changed from "block inline-box replaced" to
the default, "block inline replaced". This will allow us to pass webvtt web platform
test class_font_shorthand.html and probably others as well.

class_font_shorthand.html tests a cue with a nested <c> element, where <c> is given a line height of 18px.
Because of  -webkit-line-box-contain:"block inline-box replaced", the line height of the cue is not
impacted by the line height of <c>, causing it to differ from the reference file.

* Source/WebCore/Modules/modern-media-controls/controls/text-tracks.css:
(video::-webkit-media-text-track-container):

Canonical link: https://commits.webkit.org/301157@main
